bentoml/_internal/io_descriptors/pandas.py

```diff
@@ -182,15 +179,144 @@ async def to_http_response(self, obj: "pd.DataFrame") -> Response:
-        enforce_dtype: bool = True,
         enforce_shape: bool = True,
+        enforce_dtype: bool = False,
     ) -> "PandasDataFrame":
-        """Create an IO Descriptor from given inputs."""
+        """
+        Create a PandasDataFrame IO Descriptor from given inputs.
+
+        Args:
+            sample_input (`pd.DataFrame`):
+                Given inputs
+            orient (`str`, `optional`, default to `records`):
+                Indication of expected JSON string format. Compatible JSON strings can be produced
+                 by `pandas.io.json.to_json()` with a corresponding orient value. Possible orients are:
+                    - `split` - `Dict[str, Any]`: {idx -> [idx], columns -> [columns], data -> [values]}
+                    - `records` - `List[Any]`: [{column -> value}, ..., {column -> value}]
+                    - `index` - `Dict[str, Any]`: {idx -> {column -> value}}
+                    - `columns` - `Dict[str, Any]`: {column -> {index -> value}}
+                    - `values` - `Dict[str, Any]`: Values arrays
+            apply_column_names (`bool`, `optional`, default to `True`):
+                Update incoming DataFrame columns. `columns` must be specified at function signature.
+                 If you don't want to enforce a specific columns name then change `apply_column_names=False`.
+            enforce_dtype (`bool`, `optional`, default to `True`):
+                Enforce a certain data type. `dtype` must be specified at function signature.
+                 If you don't want to enforce a specific dtype then change `enforce_dtype=False`.
+            enforce_shape (`bool`, `optional`, default to `False`):
+                Enforce a certain shape. `shape` must be specified at function signature.
+                 If you don't want to enforce a specific shape then change `enforce_shape=False`.
+
+        Returns:
+            `PandasDataFrame` IODescriptor from given users inputs.
+
+        Examples::
+            import pandas as pd
+            from bentoml.io import PandasDataFrame
+            arr = [[1,2,3]]
+            inp = PandasDataFrame.from_sample(pd.DataFrame(arr))
+
+            ...
+            @svc.api(input=inp, output=PandasDataFrame())
+            def predict(inputs: pd.DataFrame) -> pd.DataFrame:...
+        """
+        columns = [str(x) for x in list(sample_input.columns)]
+        return cls(
+            orient=orient,
+            enforce_shape=enforce_shape,
+            shape=sample_input.shape,
+            apply_column_names=apply_column_names,
+            columns=columns,
+            enforce_dtype=enforce_dtype,
+            dtype=None,  # TODO: not breaking atm
+        )
+
+
+_repl = {"PandasDataFrame": "PandasSeries", "DataFrame": "Series"}
+_rgx = re.compile("|".join(dict((re.escape(k), v) for k, v in _repl.items()).keys()))
+
+
+class PandasSeries(PandasDataFrame):
+    __doc__ = _rgx.sub(lambda m: _repl[re.escape(m.group(0))], PandasDataFrame.__doc__)
```

Comment:
i'd suggest copying over the docs here, especially we want to add some sample client HTTP requests and they will be different between these two

Reply:
noted